mobile/native/src/emergency_kit.rs

```diff
@@ -44,6 +52,86 @@ pub fn delete_position() -> Result<()> {
     Ok(())
 }
 
+pub fn recreate_position() -> Result<()> {
+    tracing::warn!("Executing emergency kit! Recreating position!");
+    let node = get_node();
+    let counterparty = config::get_coordinator_info().pubkey;
+    let channel = node.inner.get_signed_channel_by_trader_id(counterparty)?;
+
+    ensure!(
+        matches!(channel.state, SignedChannelState::Established { .. }),
+        "A position can only be recreated from an established signed channel state"
+    );
+
+    let positions = db::get_positions()?;
+    let position = positions.first();
+    ensure!(
+        position.is_none(),
+        "Can't recreate a position if there is already a position"
+    );
+
+    let order = db::get_last_failed_order()?.context("Couldn't find last failed order!")?;
+    let average_entry_price = order.execution_price().context("Missing execution price")?;
+
+    tracing::debug!("Creating position from established signed dlc channel and last failed order");
+
+    let contract_id = channel.get_contract_id().context("Missing contract id")?;
+
+    let contract = node
+        .inner
+        .get_contract_by_id(&contract_id)?
+        .context("Missing contract")?;
+
+    let (collateral, expiry) = match contract {
+        Contract::Signed(contract) | Contract::Confirmed(contract) => {
+            let oracle_event = &contract
+                .accepted_contract
+                .offered_contract
+                .contract_info
+                .first()
+                .context("missing contract info")?
+                .oracle_announcements
+                .first()
+                .context("missing oracle info")?
+                .oracle_event;
+
+            let expiry_timestamp =
+                OffsetDateTime::from_unix_timestamp(oracle_event.event_maturity_epoch as i64)?;
+
+            (
+                contract.accepted_contract.accept_params.collateral,
```

Comment:
🤔 This is the total DLC collateral, which does not take into account the collateral reserve that is not being traded. I think the position's `collateral` is actually meant to be the trading margin, which can be calculated like this:

```
contract.accepted_contract.accept_params.collateral - trader_collateral_reserve
```

The reserve is given by `get_usable_dlc_channel_balance`.